packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_loan_payback_loan.test.ts

```diff
@@ -0,0 +1,770 @@
+    await expect(promise).rejects.toThrow(DeFiDRpcError)
+    await expect(promise).rejects.toThrow(`Cannot find existing vault with id ${'0'.repeat(64)}`)
+  })
+
+  it('should not paybackLoan on nonexistent loan token', async () => {
+    const txn = await bBuilder.loans.paybackLoan({
+      vaultId: bobVaultId,
+      from: bobColScript,
+      tokenAmounts: [{ token: 1, amount: new BigNumber(1) }]
+    }, bobColScript)
+
+    const promise = sendTransaction(bob.container, txn)
+    await expect(promise).rejects.toThrow(DeFiDRpcError)
+    await expect(promise).rejects.toThrow('Loan token with id (1) does not exist!')
+  })
+
+  it('should not paybackLoan as no loan on vault', async () => {
+    const txn = await bBuilder.loans.paybackLoan({
+      vaultId: bobVaultId1,
+      from: bobColScript,
+      tokenAmounts: [{ token: 2, amount: new BigNumber(30) }]
+    }, bobColScript)
+
+    const promise = sendTransaction(bob.container, txn)
+    await expect(promise).rejects.toThrow(DeFiDRpcError)
+    await expect(promise).rejects.toThrow(`There are no loans on this vault (${bobVaultId1})`)
+  })
+
+  it('should not paybackLoan as no token in this vault', async () => {
+    const txn = await bBuilder.loans.paybackLoan({
+      vaultId: bobVaultId,
+      from: bobColScript,
+      tokenAmounts: [{ token: 3, amount: new BigNumber(30) }]
+    }, bobColScript)
+
+    const promise = sendTransaction(bob.container, txn)
+    await expect(promise).rejects.toThrow(DeFiDRpcError)
+    await expect(promise).rejects.toThrow('There is no loan on token (AMZN) in this vault!')
+  })
+
+  it('should not paybackLoan on empty vault', async () => {
+    const emptyVaultId = await bob.rpc.loan.createVault({
+      ownerAddress: bobVaultAddr,
+      loanSchemeId: 'scheme'
+    })
+    await bob.generate(1)
+
+    const txn = await bBuilder.loans.paybackLoan({
+      vaultId: emptyVaultId,
+      from: bobColScript,
+      tokenAmounts: [{ token: 2, amount: new BigNumber(30) }]
+    }, bobColScript)
+
+    const promise = sendTransaction(bob.container, txn)
+    await expect(promise).rejects.toThrow(DeFiDRpcError)
+    await expect(promise).rejects.toThrow(`Vault with id ${emptyVaultId} has no collaterals`)
+  })
+
+  it('should not paybackLoan on liquidation vault', async () => {
+    await alice.generate(6)
+
+    const liqVault = await bob.container.call('getvault', [bobLiqVaultId])
+    expect(liqVault.state).toStrictEqual('inLiquidation')
+
+    const txn = await bBuilder.loans.paybackLoan({
+      vaultId: bobLiqVaultId,
+      from: bobColScript,
+      tokenAmounts: [{ token: 2, amount: new BigNumber(30) }]
+    }, bobColScript)
+
+    const promise = sendTransaction(bob.container, txn)
+    await expect(promise).rejects.toThrow(DeFiDRpcError)
+    await expect(promise).rejects.toThrow('Cannot payback loan on vault under liquidation')
+  })
+
+  it('should not paybackLoan with arbitrary utxo', async () => {
```

Comment:
utxo?
```suggestion
  it('should not paybackLoan with invalid auth', async () => {
```
?

Reply:
ohh.. that's means using the other utxo which is not owned by user
yea `invalid auth` is more straight to the point